src/sagemaker/fw_utils.py

```diff
@@ -808,11 +832,52 @@ def validate_distribution(
             py_version=py_version,
             image_uri=image_uri,
         )
+        validate_torch_distributed_distribution(
+            instance_type=instance_type,
+            distribution=distribution,
+            framework_name=framework_name,
+            framework_version=framework_version,
+            py_version=py_version,
+            image_uri=image_uri,
+        )
         warn_if_parameter_server_with_multi_gpu(
             training_instance_type=instance_type, distribution=distribution
         )
     return distribution
 
+def validate_supported_distributions(
+    instance_type, distribution
+):
+    """Check if the provided distribution strategy is supported for the instance_type
+
+    Args:
+        instance_type (str): A string representing the type of training instance selected.
+        distribution (dict): A dictionary with information to enable distributed training.
+    """
+    match = re.match(r"^ml[\._]([a-z\d]+)\.?\w*$", instance_type)
+    err_msg = ""
+    if match and match[1].startswith("trn"):
+        keys = distribution.keys()
+        if len(keys) == 0:
+            return
+        elif len(keys) == 1:
+            distribution_strategy = keys[0]
+            if distribution_strategy != "torch_distributed":
+                err_msg += (
+                f"Provided distribution strategy {distribution_strategy} is not supported by"
+                " Trainium instances yet.\n"
```

Comment:
Unless we guarantee to support all, I think we should remove the "yet" from this error message (and one below.)

Reply:
+1

Reply:
Updated the wording.